### ISSUE TITLE
Hls adaptive bitrate

### ISSUE THREAD

**Issue by minhtoantm** [MrQiuHaHa/ijkplayer_iOS#1]
Can you implement hls adaptive bitrate?
I tried but nơ I don't have any solution for this function.
Thank you.

**Comment by MrQiuHaHa** [comment]
Sorry，I don't know either.

**Comment by minhtoantm** [comment]
Thank you

**Comment by MrQiuHaHa** [comment]
Where are you from ?

**Comment by minhtoantm** [comment]
I'm from Vietnam.

**Comment by MrQiuHaHa** [comment]
HaHa，Nice to meet you ！I am Chinese

**Comment by minhtoantm** [comment]
> HaHa，Nice to meet you ！I am Chinese

Yeah, I knew it from your profile.
Nice to meet you too

**Comment by MrQiuHaHa** [comment]
> > HaHa，Nice to meet you ！I am Chinese
> 
> Yeah, I knew it from your profile.
> Nice to meet you too

Are you an iOS developer ?

**Comment by minhtoantm** [comment]
I can develop both of iOS and Android, but not professional like you. :D

**Comment by MrQiuHaHa** [comment]
Haha, You're too modest. I've been working for five years. My salary is ok, but I need to pay the  house loan and car loan. Living in China is quite stressful. I don't know if programmers in other countries are the same. Because I don't know any foreign programmer friends.

**Comment by minhtoantm** [comment]
You have just been working for five years and now you have both house and car. So amazing.
In Vietnam, house or car is big property and it's too hard to bye a house or a car by yourself without support from your family.

**Comment by MrQiuHaHa** [comment]
Haha, So it looks good to China. My family is not rich, I buy a house and a car only by myself, and I plan to get a new car next month. IT seems that I would like to thank my work and study IT harder in the future. Thank you and good luck in your work ！

**Comment by minhtoantm** [comment]
get a new car next month --> so amazing...

BTW, can you share me some website like github on China? I want to research this issue in China because Chinese Developers are so good in video or graphic.

**Comment by MrQiuHaHa** [comment]
Do you want to adapt the bit rate to the network state as the video plays？Ijkplayer is a very popular video library in China, and it is open source. You can ask questions on his official Github page. Ijkplayer should be able to set the start rate adaptive. Experts in audio and video technology are very valuable in China. It's a pity That I'm not an expert in this field, or I could help you. [https://github.com/bilibili/ijkplayer](url)

**Comment by minhtoantm** [comment]
> Do you want to adapt the bit rate to the network state as the video plays?

Yes. But the first function what I need is "change stream without flag". I tried to ask question in ijkplayer github but nobody replies.
Moreover, bilibili no longer maintenance ijkplayer.
So I tried to ask some developer who continue fork and developing ijkplayer, but not found any solution in English.
I try to search by Chinese but actually I don't know Chinese, I tried to use Google Translate but no helpful...
Can you tell me some keywords in Chinese for this issue?
